posthog/batch_exports/sql.py

```diff
@@ -336,3 +336,151 @@
 ORDER BY interval_start desc
 SETTINGS max_replica_delay_for_distributed_queries=1
 """
+
+CREATE_SESSIONS_BATCH_EXPORT_VIEW = f"""
+CREATE OR REPLACE VIEW sessions_batch_export ON CLUSTER {settings.CLICKHOUSE_CLUSTER} AS (
+    SELECT
+        team_id as team_id,
+        session_id_v7 as session_id_v7,
+        argMaxMerge(distinct_id) as distinct_id,
```

Comment:
maybe you could take me through the logic in this view just so I can get a bit of context behind it? for example, I'm not too familiar with some of these functions, such as `argMaxMerge`, and would like to understand why we're using them and why we sometimes use `argMaxMerge` vs `argMinMerge` vs `max`, etc

Reply:
I just copied it over from the existing sessions views.

Reply:
This SQL is a copy of this view: https://github.com/PostHog/posthog/blob/master/posthog/models/raw_sessions/sql.py#L448. But with the batch export filters added.

Reply:
Looking at it, I seem to have forgotten about:
```
fromUnixTimestamp(intDiv(toUInt64(bitShiftRight(session_id_v7, 80)), 1000)) as session_timestamp,
```
Maybe that should be our batch export timestamp.

Reply:
But it does raise the question of when the UUID is set: If it's set before ingesting the data into clickhouse, we could run into problems with replication being delayed in clickhouse (again).

Reply:
> This SQL is a copy of this view: master/posthog/models/raw_sessions/sql.py#L448. But with the batch export filters added.

Thanks. Could we just query that view directly with our batch export filters then rather than creating another view?

Reply:
> Thanks. Could we just query that view directly with our batch export filters then rather than creating another view?

Not really: The view runs the aggregations without filters. If we add filters after the view, we are only filtering the aggregated results. For performance, we want to filter the data _before_ it's aggregated.

Reply:
Ok, that makes sense 👍 But there's still the question of whether we should create a view at all, since we were trying to move away from creating views for batch exports. (I suppose the view is useful for encapsulating the logic and reusing it when debugging, etc)

Reply:
Quick update: I am now selecting everything from the `sessions` HogQL table, so the schema matches 1:1 with what users can query in the data warehouse `sessions` table. I liked this for consistency with other parts of PostHog.